### ISSUE TITLE
[HOLD for payment 2024-08-14] [$250] [CVP] [Critical] Not redirecting to workspace chat and not converting the IOU to an expense report when Pay with business account selected

### ISSUE THREAD

**Issue by m-natarajan** [Expensify/App#46339]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.13-3
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** @quinthar 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1722037742346479

## Action Performed:

1. User A submits an IOU to User B
2. User B selects option "Pay with expensify"
3. Select option "business bank account"
4. Observe the LHN 

## Expected Result:
A new workspace created and moved the IOU to "expense report" for the payer and user redirected to workspace chat

## Actual Result:
Workspace created , Not redirected to workspace chat and not converting the IOU to an expense report
## Workaround:
Unknown
## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

https://github.com/user-attachments/assets/75f3f363-e519-41e9-942e-423149c1921c

![image (2)](https://github.com/user-attachments/assets/e42f81af-18dc-44f3-9774-89115411e73e)

https://github.com/user-attachments/assets/db2e3217-a70b-476a-8bb3-b6c6db6d6192

[Logs.txt](https://github.com/user-attachments/files/16397750/Logs.txt)

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01f077fda3c01ec8f0</li>
        <li>Upwork Job ID: 1818013143723474571</li>
        <li>Last Price Increase: 2024-07-29</li>
<li>Automatic offers: </li>
<ul>
<li>gijoe0295 | Contributor | 103331425</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @trjExpensify</details>

**Comment by trjExpensify** [comment]
Confirming a couple of things [in thread](https://expensify.slack.com/archives/C049HHMV9SM/p1722251759353779?thread_ts=1722037742.346479&cid=C049HHMV9SM). 

In the meantime, @gijoe0295 can you include a video of your proposal above to check the expected results? Thanks!

**Comment by trjExpensify** [comment]
Okay, so sounds like the two items to look at here are:

- Redirect to the workspace chat when clicking the “Pay with business bank account” button option right away when we create the workspace, and open the add bank account page in the RHP of the workspace chat not the DM.
- Make sure the workspace chat with the converted expense report is GBR’d right away, it doesn't require a click into the report preview for it to appear on the workspace chat row in the LHN.

**Comment by trjExpensify** [comment]
@mananjadhav can you jump on that [proposal review](https://github.com/Expensify/App/issues/46339#issuecomment-2254183898), please? We'd love to get this one turned around ASAP. Thanks!

**Comment by trjExpensify** [comment]
^^ @marcaaron given the context on that PR that deployed a few hours ago, checking you're okay with that change. 

For the bottom-up flow, both of these options should be retained: 

<img width="923" alt="image" src="https://github.com/user-attachments/assets/50b94e01-f2b0-4cc3-a7b9-324c11dfb790">

**Comment by mananjadhav** [comment]
@gijoe0295 's proposal looks good to me.

🎀 👀 🎀 C+ reviewed.

**Comment by mananjadhav** [comment]
@danieldoglas Should we handle [this](https://github.com/Expensify/App/issues/46339#issuecomment-2257023755) within this PR scope? or separately as a regression? I can see the revert for the linked PR was closed.

**Comment by trjExpensify** [comment]
That issue is linked to a [deploy blocker](https://github.com/Expensify/App/issues/46490), we might need to CP a fix for that. Let's discuss here though: https://expensify.slack.com/archives/C01GTK53T8Q/p1722351533575289

**Comment by marcaaron** [comment]
>checking you're okay with that change

Yep, my intention was to remove only the debit card as payment option.

**Comment by trjExpensify** [comment]
Yeah, we ended up doing two PRs in the end to clear that blocker?

**Comment by mananjadhav** [comment]
@trjExpensify We implemented two PRs - one for this blocker which I wrote and the second one for another issue that [this PR](https://github.com/Expensify/App/issues/46339#issuecomment-2288363410) @gijoe0295 wrote.

But yeah I agree they posted the solution before I implemented in the comments. I implemented it unblock the release.

**Comment by trjExpensify** [comment]
Cool, so I'll do the following:

- $375 to @gijoe0295 for this issue + taking us most of the way with the proposal on the other issue. 
- $500 to @mananjadhav for the other issue's PR, and the C+ review of this one.
- $250 to @mollfpr for the C+ review of the other issue's PR

How does that sound?

**Comment by mananjadhav** [comment]
A little confused here. Are we talking about any other issue? I don’t think the blocker PR implemented was reviewed by C+. I am not in front of the system. I can check later.

**Comment by mollfpr** [comment]
@trjExpensify I don't think I reviewed any PR regarding this issue 😅

**Comment by trjExpensify** [comment]
Ah, @mollfpr is right. You were assigned, but didn't review. Cool, so two issues, two PRs: 

Issue: [Not redirecting to workspace chat and not converting the IOU to an expense report when Pay with business account selected](https://github.com/Expensify/App/issues/46339)
- [PR](https://github.com/Expensify/App/pull/46557): Authored by @gijoe0295, reviewed by @mananjadhav. 

Issue: [Expense - "Business bank account" option appears briefly when paying with Expensify](https://github.com/Expensify/App/pull/46598)
- [PR](https://github.com/Expensify/App/pull/46598): Authored by @mananjadhav, reviewed by Rory (internal). 

So amending the payment summary as follows:

- $375 to @gijoe0295 for this issue + taking us most of the way with the proposal on the other issue. 
- $500 to @mananjadhav for the other issue's PR, and the C+ review of this one.

Sound right?

**Comment by mananjadhav** [comment]
That’s right.

**Comment by trjExpensify** [comment]
Okay great, go ahead and request Manan. @gijoe0295 I've settled up.

**Comment by JmillsExpensify** [comment]
$500 approved for @mananjadhav
